core/src/main/java/org/mapfish/print/processor/map/CreateMapProcessor.java

```diff
@@ -289,12 +289,12 @@ private MapfishMapContext getTransformer(final MapfishMapContext mapContext, fin
         //now we download and cache all images at once
         cache.cache(this.requestForkJoinPool);
 
-        for (int i = 0; i < layers.size(); i++) {
-            MapLayer layer = layers.get(i);
+        for (int i = 0; i < layers.size();) {
```

Comment:
I it possible to split this big loop in 2, where the first one calculate the layers to merge...

Reply:
OK, done